pkgs/servers/mir/default.nix

```diff
@@ -0,0 +1,129 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, boost
+, egl-wayland
+, libglvnd
+, glm
+, glog
+, lttng-ust
+, udev
+, glib
+, wayland
+, libxcb
+, xorg
+, libdrm
+, mesa
+, libepoxy
+, nettle
+, libxkbcommon
+, libinput
+, libxmlxx
+, libuuid
+, pcre2
+, freetype
+, libyamlcpp
+, python3
+, libevdev
+, libselinux
+, libsepol
+, eglexternalplatform
+, doxygen
+, libxslt
+, pkg-config
+}:
+
+let
+  pythonEnv = python3.withPackages(ps: [ ps.pillow ]);
+in
+
+stdenv.mkDerivation rec {
+  pname = "mir";
+  version = "2.10.0";
+
+  src = fetchFromGitHub {
+    owner = "MirServer";
+    repo = "mir";
+    rev = "v${version}";
+    sha256 = "sha256-HPbToo7lhsHyx0w4oO2wp3zZ0xpr6SlEm2oE6Rm8b3E=";
+  };
+
+  postPatch = ''
+    # Fix error broken path found in pc file ("//")
+    for f in $(find . -name '*.pc.in') ; do
+      substituteInPlace $f \
+        --replace "/@CMAKE_INSTALL_LIBDIR@" "@CMAKE_INSTALL_LIBDIR@" \
+        --replace "/@CMAKE_INSTALL_INCLUDEDIR@" "@CMAKE_INSTALL_INCLUDEDIR@"
+    done
```

Comment:
While trying to build something that depends on Mir we noticed that the paths are still broken, just not broken enough for the pkg-config check to notice:

```
prefix=/nix/store/yr3jc3213w1y3f1kc7ppa1a5yx0xng4y-mir-2.10.0
libdir=${prefix}/nix/store/yr3jc3213w1y3f1kc7ppa1a5yx0xng4y-mir-2.10.0/lib
includedir=/nix/store/vzy7w5yqi6y5z5qjpiyq1yfpjpxkzk6i-mir-2.10.0-dev/nix/store/vzy7w5yqi6y5z5qjpiyq1yfpjpxkzk6i-mir-2.10.0-dev/include/miral
```

I think for our purposes, replacing all `${prefix}/@CMAKE_INSTALL_<dir>@` with `@CMAKE_INSTALL_FULL_<dir>@` etc might be fine:

```
prefix=/nix/store/q4921w10hjmpk28dkk1c4y1c5fy33fcy-mir-2.10.0
libdir=/nix/store/q4921w10hjmpk28dkk1c4y1c5fy33fcy-mir-2.10.0/lib
includedir=/nix/store/ird024yncpxv3rlz42grx8i4g3s891s4-mir-2.10.0-dev/include/miral
```

The more proper fix being something like [jtojnar/cmake-snips#concatenating-paths-when-building-pkg-config-files](https://github.com/jtojnar/cmake-snips/tree/a9d72829ab0cb42e541c9975e698d6b8282610ed#concatenating-paths-when-building-pkg-config-files) being done upstream.

Reply:
> being done upstream.

Should we `@-`mark them?

Reply:
I PR'd the same fix to their wlcs project recently (first with `FULL`, now with jtojnar's snippet), guess I could do the same for Mir & link it here for `fetchpatch`ing.

Reply:
https://github.com/MirServer/mir/pull/2786, should be cherry-pickable onto 2.11.0.